### PR TITLE
INT-3699: RemoteFileTemplate Javadocs

### PR DESCRIPTION
**Pull request by garyrussell**
JIRA: https://jira.spring.io/browse/INT-3699

Also javadocs for related classes.

### CODE REVIEW

File: spring-integration-file/src/main/java/org/springframework/integration/file/remote/RemoteFileTemplate.java [hunk 1]
```diff
@@ -87,60 +89,120 @@
...
+	 * @param temporaryRemoteDirectoryExpression the temporary remote directory expression.
+	 */
 	public void setTemporaryRemoteDirectoryExpression(Expression temporaryRemoteDirectoryExpression) {
 		Assert.notNull(temporaryRemoteDirectoryExpression, "temporaryRemoteDirectoryExpression must not be null");
 		this.temporaryDirectoryExpressionProcessor = new ExpressionEvaluatingMessageProcessor<String>(temporaryRemoteDirectoryExpression, String.class);
 	}
 
+	/**
+	 * Set the file name expression to determine the name of the local file when retrieving
+	 * a remote file.
```
Comment (artembilan) [inline spring-integration-file/src/main/java/org/springframework/integration/file/remote/RemoteFileTemplate.java]: Something wrong is here:
```
* Set the expression for remote file to retrieve.
```
?
````java
public boolean get(Message<?> message, InputStreamCallback callback) {
	Assert.notNull(this.fileNameProcessor, "A 'fileNameExpression' is needed to use get");
	String remotePath = this.fileNameProcessor.processMessage(message);
	return this.get(remotePath, callback);
}
````
It is exactly a `remote path`. There is nothing about `local`

File: spring-integration-file/src/main/java/org/springframework/integration/file/remote/RemoteFileTemplate.java [hunk 2]
```diff
@@ -87,60 +89,120 @@
...
+	 * Set whether a temporary file name is used when sending files to the remote system.
+	 * @param useTemporaryFileName true to use a temporary file name.
+	 * @see #setTemporaryFileSuffix(String)
+	 */
 	public void setUseTemporaryFileName(boolean useTemporaryFileName) {
 		this.useTemporaryFileName = useTemporaryFileName;
 	}
 
+	/**
+	 * Set the file name generator used to determine the remote filename to be used when transferring
```
Comment (artembilan) [inline spring-integration-file/src/main/java/org/springframework/integration/file/remote/RemoteFileTemplate.java]: s/determine/generate

**Comment by artembilan** [comment]
Let me know and I'll fix on merge

**Comment by garyrussell** [comment]
Pushed

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-integration/commit/15150ac49917de45acc3c8676e6dc5dd3efe8b4b after squesh

**Pull request closed without merging**